src/features/home/view/first/home-first-search-box.components.tsx

```diff
@@ -0,0 +1,151 @@
+import { AutoComplete, Button, Form, Input, Rate, Spin } from "antd";
+import useTranslation from "next-translate/useTranslation";
+import { useRouter } from "next/router";
+import { useContext, useEffect, useState } from "react";
+import { AppContext } from "../../../../core/app/app.context";
+import { PagesUrls, TranslationFiles } from "../../../../core/core";
+import { SearchResultsRestaurants } from "../../../restaurants/services/restaurants/models/search-results-restaurants.models";
+import { restaurantsServices } from "../../../restaurants/services/restaurants/restaurants.services";
+import useDebounce from "../../../shared/hooks/debounce.hooks";
+import classes from "./style.module.css";
+
+const HomeFirstSearchBox = () => {
+  const { direction } = useContext(AppContext);
+
+  const { t } = useTranslation(TranslationFiles.HOME);
+
+  const [form] = Form.useForm();
+
+  const [restaurantName, setRestaurantName] = useState("");
+
+  const [searchResultsForRestaurants, setSearchResultsForRestaurants] =
+    useState<SearchResultsRestaurants[]>([]);
+
+  const [options, setOptions] = useState<any>([]);
+
+  const debouncedSearchTerm = useDebounce(restaurantName, 500);
+
+  const { push } = useRouter();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (debouncedSearchTerm) {
+      getRestaurants(restaurantName);
+    }
+  }, [debouncedSearchTerm]);
+
+  const getRestaurants = async (value: any) => {
+    setIsLoading(true);
+    try {
+      let result = await restaurantsServices.getAll({
+        size: 5,
+        query: { fuzzy: { keywords: { value: value } } },
+      });
+      setSearchResultsForRestaurants(result.hits.hits);
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const searchResults = () => {
+    let option: any[] = [];
+    searchResultsForRestaurants.map((item: SearchResultsRestaurants) => {
+      option.push({
+        value: item._source.name,
+        label: (
+          <div
+            style={{
+              display: "flex",
+              justifyContent: "space-between",
+            }}
+          >
+            <span>{item._source.name}</span>
+            <span>
+              <span style={{ margin: "0px 15px" }}>
+                <Rate
+                  disabled
+                  style={{ fontSize: "18px" }}
+                  value={item._source.rated_by}
+                  allowHalf={true}
+                />
+              </span>
+              <Button
+                type="ghost"
+                onClick={() => {
+                  push(`${PagesUrls.RESTAURANTS}/${item._source.name}`);
+                }}
+              >
+                {t("first.show-restaurant")}
+              </Button>
+            </span>
+          </div>
+        ),
+      });
+    });
+    return option;
+  };
+
+  useEffect(() => {
+    setOptions(searchResults());
```

Comment:
Here we can put content of "searchResults" function in "useEffect" directly or use "useCallback" to declare a "searchResults" function